### PR TITLE
Ap 500 new capital vehicle questions for providers

### PR DESCRIPTION
**Pull request by reggieb**
[Jira AP-500](https://dsdmoj.atlassian.net/browse/AP-500)

Add facility to submit more detailed vehicle information.

## Changes

- Adds Vehicles controller - to create a vehicle when told applicant has one
- Adds `vehicles` name space for controllers that modify vehicle.
- Adds facility to prefix step names to avoid duplication
- Within `vehicles` namespace adds Vehicles attribute controllers: Estimated Value, Remaining Payments, Purchase Date and Regular Use

### CODE REVIEW

File: app/services/flow/flows/vehicle_steps.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module Flow
+  module Flows
+    class VehicleSteps < FlowSteps
```
Comment (reggieb) [inline app/services/flow/flows/vehicle_steps.rb]: The vehicle steps pushed `ProviderCapital` flow class over the size limit recommended by Rubocop - so split the vehicle steps into their own flow.

File: spec/requests/providers/vehicles/purchase_dates_spec.rb [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    end
+
+    context 'without a value' do
+      let(:params) do
+        {
+          vehicle: {
+            purchased_on_year: purchase_date.year,
+            purchased_on_month: '',
+            purchased_on_day: purchase_date.day
+
```
Comment (Kalagan) [inline spec/requests/providers/vehicles/purchase_dates_spec.rb]: remove the extra empty line?

File: spec/requests/providers/vehicles/purchase_dates_spec.rb [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      end
+
+      context 'with blank entry' do
+        let(:params) do
+          {
+            vehicle: {
+              purchased_on_year: '',
+              purchased_on_month: '',
+              purchased_on_day: ''
+
```
Comment (Kalagan) [inline spec/requests/providers/vehicles/purchase_dates_spec.rb]: remove the extra empty line?

File: app/controllers/providers/vehicles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      else
+        vehicle.errors.add :exists_yes, I18n.t('providers.vehicles.show.nothing_selected')
+        render :show
+      end
+    end
+
+    private
+
+    def create_vehicle_and_continue
+      vehicle.save unless vehicle.persisted?
```
[the remark this answers is not in the document]
Comment (reggieb) [reply app/controllers/providers/vehicles_controller.rb]: Must get into the habit of using `save!`. Too many years of using the default.

File: app/views/providers/vehicles/regular_uses/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        url: providers_legal_aid_application_vehicles_regular_use_path,
+        method: :patch,
+        local: true
+      ) do |form| %>
+    <%= govuk_form_group show_error_if: @form.errors.present? do %>
+      <%= govuk_fieldset_header page_heading, padding_below: 0 %>
+      <%= govuk_hint t('.include_partner_use') %>
+      <div class="govuk-radios govuk-radios--conditional govuk-!-padding-bottom-6 govuk-!-padding-top-2"
+           data-module="radios">
+        <%= form.govuk_radio_button :used_regularly, 'true', label: t('generic.yes') %>
```
[the remark this answers is not in the document]
Comment (reggieb) [reply app/views/providers/vehicles/regular_uses/show.html.erb]: No I didn't .... and yes I should have.

**Pull request merged**